### PR TITLE
♻️ #28966: Replaced Array usage with Set usage, where appropriate.

### PR DESCRIPTION
**Pull request by xiexr151e**
This is a subset of the fix for issue #28966.
Because refactoring Map will be much more involved (there were 70+ instances of Map usage found, and changing to Map may break current unit tests), refactoring for Map usage should be its own PR.

### CODE REVIEW

File: src/polyfillstub/intersection-observer-stub.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ export class IntersectionObserverStub {
     if (this.inst_) {
       this.inst_.disconnect();
     } else {
-      this.elements_.length = 0;
+      this.elements_.clear();
```
Comment (xiexr151e) [inline src/polyfillstub/intersection-observer-stub.js:165]: I am uncertain as to if there is a better way to modify this line. @jridgewell

File: src/polyfillstub/intersection-observer-stub.js [hunk 2]
```diff
@@ -112,8 +112,8 @@ export class IntersectionObserverStub {
       throw new Error('root must be an Element');
     }
 
-    /** @private {?Array<!Element>} */
-    this.elements_ = [];
+    /** @private {?Set<!Element>} */
+    this.elements_ = new Set();
```
Comment (xiexr151e) [inline src/polyfillstub/intersection-observer-stub.js:116]: We don't know if elements in `this.elements_` are unique ahead of time. Should I therefore keep this as an Array? @jridgewell

File: extensions/amp-subscriptions/0.1/platform-store.js [hunk 3]
```diff
@@ -87,8 +87,8 @@ export class PlatformStore {
     /** @private {?Deferred<!Array<!./entitlement.Entitlement>>} */
     this.allResolvedPromise_ = null;
 
-    /** @private {!Array<string>} */
-    this.failedPlatforms_ = [];
+    /** @private {!Set<string>} */
+    this.failedPlatforms_ = new Set();
```
Comment (xiexr151e) [inline extensions/amp-subscriptions/0.1/platform-store.js:91]: I'm pretty sure the elements in here should be unique.

**Comment by erwinmombay** [comment]
closing as stale

**Pull request closed without merging**